Sources/SwiftRedis/RedisResp.swift

```diff
@@ -207,30 +216,28 @@ class RedisResp {
     // Mark: Parser helper functions
 
     private func compare(_ buffer: inout Data, at offset: Int, with: Data) throws -> (Bool, Int) {
-        while  offset+with.count >= buffer.count {
+        // use offset+1 instead of offset+with.count as with.count is always == 1
+        while offset+1 >= buffer.count {
             let length = try socket?.read(into: &buffer)
             if  length == 0 {
                 throw RedisRespError(code: .EOF)
             }
         }
 
-        let range = buffer.range(of: with, options: [], in: offset..<offset+with.count)
-        if range != nil {
-            return (true, offset+with.count)
+        if (buffer[offset..<offset+1] == with){
+            return (true, offset+1)
         } else {
             return (false, offset)
         }
     }
 
     private func find(_ buffer: inout Data, from: Int, data: Data) throws -> Int {
         var offset = from
-        var notFound = true
-
-        while notFound {
+        while true {            
             let range = buffer.range(of: data, options: [], in: offset..<buffer.count)
             if range != nil {
-                offset = (range?.lowerBound)!
-                notFound = false
```

Comment:
Do you need these fixes if you are using a different `find` method?
I think it may be more interesting to keep it as it is (unless that's a bug fix?)